frontend/model/state.js

```diff
@@ -157,7 +207,37 @@ sbp('sbp/selectors/register', {
           upgradeAction(contractID)
         })
       }
-    })()
+    })
+    contractUpdate((contractIDHints: ?string[]) => {
+      const state = sbp('state/vuex/state')
+      if (Array.isArray(contractIDHints)) {
+        if (!contractIDHints.reduce((acc, contractID) => {
+          return (acc || state.contracts[contractID]?.type === 'gi.contracts/group')
+        }, false)) return
+      }
+      // Update expired invites
+      // If fewer than MAX_GROUP_MEMBER_COUNT 'anyone can join' have been used,
+      // create a new 'anyone can join' link up to MAX_GROUP_MEMBER_COUNT invites
+      const ourIdentityContractId = state.loggedIn?.identityContractID
+      if (!ourIdentityContractId || !state[ourIdentityContractId]?.groups) return
+      Object.entries(state[ourIdentityContractId].groups).map(([groupID, { hasLeft }]: [string, Object]) => {
+        const groupState = state[groupID]
+        if (hasLeft || !groupState?.invites) return undefined
+        if (Array.isArray(contractIDHints) && !contractIDHints.includes(groupID)) return undefined
+        const hasBeenUpdated = Object.keys(groupState.invites).some(inviteId => {
+          return groupState.invites[inviteId].creatorID === INVITE_INITIAL_CREATOR &&
+          groupState._vm.invites[inviteId].expires == null
+        })
+        if (hasBeenUpdated) return undefined
+        const usedInvites = Object.keys(groupState.invites)
+          .filter(invite => groupState.invites[invite].creatorID === INVITE_INITIAL_CREATOR)
+          .reduce((acc, cv) => acc +
+        ((groupState._vm.invites[cv].initialQuantity - groupState._vm.invites[cv].quantity) || 0), 0)
+        return (usedInvites < MAX_GROUP_MEMBER_COUNT) ? groupID : undefined
```

Comment:
This incomprehensibly dense blob seems to be a DRY violation with the same / similar logic in `fixAnyoneCanJoinLink`?

Reply:
It's similar logic but it needs to be there to know whether `fixAnyoneCanJoinLink` should be called or not. Otherwise, we'd call `fixAnyoneCanJoinLink` on every group every time a tab is opened or a new group is subscribed.